ports/windows/sleep.c

```diff
@@ -41,7 +41,12 @@ void deinit_sleep(void) {
     }
 }
 
-int usleep_impl(__int64 usec) {
+// mingw and the likes provide their own usleep() which we need to
+// replace with one that can be interrupted by APC function if needed
+// (used in mp_thread_gc)
+#undef usleep
```

Comment:
Is this the right way to do things?  If `usleep` is #define'd to something then that define will be active in all other compilation units.  So only the code in this file will pick up this custom `usleep` implementation.

Why not just keep it as `usleep_impl()`?

Reply:
Yeah that was a bit strange / wrong. `usleep_impl()` was only being used in the single `usleep()` wrapper function below on msvc, it was ignored elsewhere. I needed it to be the usleep used everywhere - but this wouldn't have actually been working as you suggest.

Turns out we don't need to provide usleep in any particular global sense, it was only being used in unix/mpconfigport.h to provide `mp_hal_delay_us()` so I think it's better to directly provide `mp_hal_delay_us()` here for win32 only.